hazelcast/src/main/java/com/hazelcast/instance/GroupProperties.java

```diff
@@ -203,7 +203,31 @@
     public static final String PROP_ELASTIC_MEMORY_UNSAFE_ENABLED = "hazelcast.elastic.memory.unsafe.enabled";
     public static final String PROP_ENTERPRISE_LICENSE_KEY = "hazelcast.enterprise.license.key";
     public static final String PROP_MAP_WRITE_BEHIND_QUEUE_CAPACITY = "hazelcast.map.write.behind.queue.capacity";
+
+    /**
+     * Defines event queue size for WAN replication, only valid for Hazelcast Enterprise
+     */
     public static final String PROP_ENTERPRISE_WAN_REP_QUEUESIZE = "hazelcast.enterprise.wanrep.queuesize";
+
+    /**
+     * Defines maximum number of WAN replication events to be gathered in a batch, only valid for Hazelcast Enterprise
+     */
```

Comment:
What happens when the batch size is reached? I guess the batch is send. 

What happens when a batch is sending and the new batch grows quickly... will it wait till the previous batch is completed and potentially growing beyond its batch size (potentially leading to oome)? Or will multiple batches be send in parallel? Is the system going to apply back pressure?